transform-plugins/src/main/java/io/cdap/plugin/UnionSplitter.java

```diff
@@ -174,42 +177,45 @@ public void transform(StructuredRecord record, MultiOutputEmitter<StructuredReco
 
     Schema.Field unionSchemaField = inputSchema.getField(unionField);
     if (unionSchemaField == null) {
-      throw new IllegalArgumentException(
-        String.format("Field '%s' does not exist in the input schema.", unionField));
-    }
-    Schema unionSchema = unionSchemaField.getSchema();
-    if (unionSchema.getType() != Schema.Type.UNION) {
-      throw new IllegalArgumentException(
-        String.format("Field '%s' is not of type union, but is of type '%s'", unionField, unionSchema.getType()));
-    }
-
-    int numFields = inputSchema.getFields().size();
-    ArrayList<Schema.Field> outputFields = new ArrayList<>(numFields);
-    int i = 0;
-    int unionIndex = -1;
-    for (Schema.Field inputField : inputSchema.getFields()) {
-      if (inputField.getName().equals(unionField)) {
-        unionIndex = i;
-        outputFields.add(null);
+      collector.addFailure(String.format("Field '%s' must exist in input schema.", unionField), null)
+        .withConfigProperty(Conf.UNION_FIELD);
+    } else {
```

Comment:
if the union field is null, it doesn't make sense to go further as the output schemas returned would be invalid. This should be:

```
if (unionSchemaField == null) {
  collector.addFailure(...);
  throw collector.getOrThrowException();
}

Schema unionSchema = ...
```

this also removes a layer of nesting, making it easier to follow

Reply:
Fixed